terraformignore.go

```diff
@@ -74,22 +74,28 @@ func readRules(input io.Reader) []rule {
 	return rules
 }
 
-func matchIgnoreRule(path string, rules []rule) bool {
-	matched := false
+func matchIgnoreRule(path string, rules []rule) (bool, *rule) {
+	var matched *rule = nil
 	path = filepath.FromSlash(path)
-	for _, rule := range rules {
+	for idx := range rules {
+		rule := &rules[idx]
 		match, _ := rule.match(path)
 
 		if match {
-			matched = !rule.excluded
+			if rule.excluded {
+				return false, rule
```

Comment:
This precedence matches `git`'s behavior, in that `git` will see `!`-prefixed lines as overriding other inclusive rules.